pkg/util/filesystem/defaultfs.go

```diff
@@ -75,6 +76,68 @@ func (fs *DefaultFs) MkdirAll(path string, perm os.FileMode) error {
 	return os.MkdirAll(fs.prefix(path), perm)
 }
 
+// fixRootDirectory fixes a reference to a drive's root directory to
+// have the required trailing slash.
+func fixRootDirectory(p string) string {
+	if len(p) == len(`\\?\c:`) {
+		if os.IsPathSeparator(p[0]) && os.IsPathSeparator(p[1]) && p[2] == '?' &&
+			os.IsPathSeparator(p[3]) && p[5] == ':' {
+			return p + `\`
+		}
+	}
+	return p
+}
+
+// MkdirAllWithPathCheck checks if path exists already. If not, it creates a directory
+// named path, along with any necessary parents, and returns nil, or else returns an error.
+// Permission bits perm (before umask) are used for all directories that MkdirAll creates.
+// If path is already a directory, MkdirAll does nothing and returns nil.
+// NOTE: In case of Windows NTFS, mount points are implemented as reparse-point
+// (similar to symlink) and do not represent actual directory. Hence Directory existence
+// check for windows NTFS should not use dir.IsDir(), rather check for symlink presence.
+func MkdirAllWithPathCheck(path string, perm os.FileMode) error {
+	// Fast path: check whether path is a directory or symlink, stop with success or error.
+	dir, err := os.Stat(path)
+	if err == nil {
+		// For windows NTFS, check if the path is symlink instead of directory.
+		if dir.IsDir() || (dir.Mode()&os.ModeSymlink != 0) {
+			return nil
+		}
+		return &os.PathError{Op: "mkdir", Path: path, Err: syscall.ENOTDIR}
+	}
+	// Slow path: make sure parent exists and then call Mkdir for path.
```

Comment:
It should be safe safe to call `os.MkdirAll` here, no need to reimplement it here.